### DISCUSSION TITLE
Testing macros?

### DISCUSSION THREAD

**Discussion by katawful** [Olical/aniseed#148] [Q&A] [answered]
I have a lot of macros in its own repo that i wanna make sure that they end up as i desire (especially if i simplify or expand the macro). attempting to use aniseed-testing doesn't seem to ever run the tests

[macros repo](https://github.com/katawful/katcros-fnl)

i've tried the following in its appropriately placed testing location, and the test never runs

```fennel
(module katcros-fnl.macros.nvim.api.autocommands.macros-test
        {import-macros [[kat :katcros-fnl.macros.nvim.api.autocommands.macros]]})

(deftest macro-cle-auc!
         (t.pr= (vim.api.nvim_clear_autocmds {}) (kat.cle-auc! {})))
```

**Comment by Olical** [comment]
Hehe, `katcros`. Sorry about the issue, my first hunch is that it's a Fennel search path issue, I'm taking a look into that theory now.

Another thing that should just work in [nfnl](https://github.com/Olical/nfnl) (although it may be infeasible for you for other reasons of course).

I assume these are working fine outside of tests? (silly question, but just want to be sure)

**Reply by katawful** [reply]
yeah they work just fine, have been using them for my configs for a while. i'll see if i can use nfnl

**Reply by Olical** [reply]
I'm still going to see if I can fix this to avoid you having to churn things for no reason, it might be something simple. But please do have a look, I strongly recommend nfnl over Aniseed, it's what I wish I'd done in the first place. I just needed to learn a lot first.

**Reply by Olical** [reply]
Oh and do you get a `.lua` file output when running your tests? The test script _should_ compile your tests to Lua first, we might have some clues in there.

**Reply by katawful** [reply]
nothing compiles in the `test/` dir

**Reply by Olical** [reply]
And one more thing, can you show me how you're running your tests and where the test file is? I can't see it in your repo so I guess it's still local?

**Reply by Olical** [reply, marked as answer]
OH NO, I just worked it out I think! Try renaming both occurrences of `macros` to `macnos`

```
test
├── fnl
│   └── katcros-fnl
│       └── macnos
│           └── nvim
│               └── api
│                   └── autocommands
│                       └── macnos-test.fnl
```

It's because Aniseed is seeing the path with the word "macro" in it and deciding not to compile the fnl because it _thinks_ it's also a macro file and can not be compiled. Again, nfnl doesn't have this issue because you tag your macro files with a comment instead of relying on naming conventions.

**Reply by katawful** [reply]
EDIT: didn't refresh the page lol

it was still local. i think i solved though? i rewrote my makefile, and i also put the test folder outside of a `macros` directory. i was planning on moving that for a new repo anyways, so i guess no harm here

```
└── test
    ├── fnl
    │   └── katcros-fnl
    │       ├── macros
    │       └── nvim
    │           └── api
    │               └── autocommands
    │                   └── macros-test.fnl
    └── lua
        └── katcros-fnl
            └── nvim
                └── api
                    └── autocommands
                        └── macros-test.lua
```

```make
.PHONY: deps compile test

default: deps compile test

deps:
	scripts/dep.sh Olical aniseed origin/master

format:
	scripts/fnlfmt.sh

compile:
	deps/aniseed/scripts/compile.sh

test:
	rm -rf test/lua
	scripts/test.sh
```
